### PR TITLE
Bug 1950025: Bump the k8s.io dependencies to v1.20.6

### PR DESCRIPTION
**Pull request by timflannagan**
The following commands were run while porting the upstream PRs downstream:

git cherry-pick -Xsubtree=staging/operator-registry 866fd1bdb808b776e40f44965852441dde3cc9f6

git cherry-pick -Xsubtree=staging/operator-lifecycle-manager b077c3e88ac5ea88b6bf3a23ac459ef8f53ec73b

make vendor

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -118,6 +125,16 @@ vendor:
...
 
+sanity/src:
+	$(CONTAINER_ENGINE) run \
+		--rm \
+		-t \
+		$(CONTAINER_RUNTIME_RUN_OPTS) \
+		-w /go/src/github.com/openshift/operator-framework-olm \
+		-v $(PWD):/go/src/github.com/openshift/operator-framework-olm \
+		$(SRC_IMAGE_REPO):$(SRC_IMAGE_TAG) \
+		make sanity
```
Comment (timflannagan) [inline Makefile:136]: I wasn't able to reproduce that vendor verification check locally that we were seeing in CI, so I added a target that runs that same check in the same container environment.

**Review by ankitathomas — APPROVED** [review]
/lgtm

**Comment by timflannagan** [comment]
Hmm, it looks like that needs-rebase label is stale/stuck. May need to try closing then re-opening again.

**Review by ankitathomas — APPROVED** [review]
/lgtm

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Pull request merged**